### PR TITLE
Add wrap - A small bubblewrap like program

### PR DESCRIPTION
**Pull request by MaxHearnden**
Add wrap.c from https://github.com/fosslinux/live-bootstrap/pull/339

### CODE REVIEW

**Comment by MaxHearnden** [comment]
This program does run into https://gitlab.com/qemu-project/qemu/-/issues/871 when using qemu user mode emulation

**Review by stikonas — COMMENTED** [review]
Perhaps you can format them in the same style as other mescc-tools-extra programs?

**Comment by stikonas** [comment]
> This program does run into https://gitlab.com/qemu-project/qemu/-/issues/871 when using qemu user mode emulation

Hmm, I guess that's mostly fine and it's qemu bug anyway. This would only affect us if we want to use this wrap when trying to build another arch, e.g. riscv64. That's mostly a development scenario and we could still use bubblewrap in that case.

File: wrap.c [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  newenv_end = copy_environment(newenv_end, "BUILDDIR");
+  newenv_end = copy_environment(newenv_end, "TMPDIR");
+  newenv_end = copy_environment(newenv_end, "OPERATING_SYSTEM");
+  newenv_end[0] = "WRAPPED=yes";
+  newenv_end[1] = NULL;
+
+
+#ifdef __M2__
+#if __uefi__
+  return spawn (argv[1], argv + sizeof(char *), newenv);
```
Comment (stikonas) [inline wrap.c:205]: Oh nice, you even made it transparent for UEFI! Just curious, have you tried running it there?

Comment (stikonas) [reply wrap.c:205]: Oh I guess you did, I can see your UEFI changes in another PR :).

**Review by stikonas — COMMENTED** [review]
looks good here but you need to rebase on top of latest commit. There are some non-trivial makefile fixes that will also affect your changes.

**Review by stikonas — APPROVED** [review]

**Pull request merged**